arc_cache.go

```diff
@@ -34,15 +34,15 @@ func newARCCachedBS(ctx context.Context, bs Blockstore, lruSize int) (*arccache,
 }
 
 func (b *arccache) DeleteBlock(k *cid.Cid) error {
-	if has, ok := b.hasCached(k); ok && !has {
+	if has, _, ok := b.hasCached(k); ok && !has {
 		return ErrNotFound
 	}
 
 	b.arc.Remove(k) // Invalidate cache before deleting.
 	err := b.blockstore.DeleteBlock(k)
 	switch err {
 	case nil, ds.ErrNotFound, ErrNotFound:
-		b.addCache(k, false)
+		b.addCache(k, 0)
```

Comment:
So, we can actually have zero-sized blocks. We'll eventually "inline" these into CIDs but that's a work in progress. You'll probably need to use -1 to indicate a missing block.

Reply:
@Stebalien
Done.  I pushed out these changes and squashed into a single commit.